bgpd/bgp_io.c

```diff
@@ -447,9 +447,9 @@ static uint16_t bgp_read(struct peer *peer)
 	size_t readsize; // how many bytes we want to read
 	ssize_t nbytes;  // how many bytes we actually read
 	uint16_t status = 0;
-	static uint8_t ibw[BGP_MAX_PACKET_SIZE * BGP_READ_PACKET_MAX];
+	uint8_t ibw[peer->max_packet_size * BGP_READ_PACKET_MAX];
 
-	readsize = MIN(ringbuf_space(peer->ibuf_work), sizeof(ibw));
+	readsize = MIN(ringbuf_space(peer->ibuf_work), peer->max_packet_size);
```

Comment:
When we are deal with 65K buffers then is it good have BGP_READ_PACKET_MAX reduced to some less number?